### PR TITLE
Configure metadata cache when using blocks storage.

### PR DESCRIPTION
**Pull request by pstibrany**
Cache is used by both querier (even if store-gateway is disabled) and store-gateway.

### CODE REVIEW

File: cortex/tsdb.libsonnet [hunk 1]
```diff
@@ -189,7 +199,7 @@
       // Persist ring tokens so that when the store-gateway will be restarted
       // it will pick the same tokens
       'experimental.store-gateway.tokens-file-path': '/data/tokens',
-    } + (if $._config.store_gateway_enabled then $.blocks_chunks_caching_config else {}),
+    } + $.blocks_chunks_caching_config + $.blocks_metadata_caching_config,
```
Comment (pracucci) [inline cortex/tsdb.libsonnet:202]: Chunks caching config should be specified only if store gateway is **disabled** or am I wrong?

Comment (pstibrany) [reply cortex/tsdb.libsonnet:202]: This is `store_gateway_args`, if store gateway is disabled, this doesn't get used.

Comment (pracucci) [reply cortex/tsdb.libsonnet:202]: Damn, I got confused during a quick look at the PR. Sorry!

Comment (pstibrany) [reply cortex/tsdb.libsonnet:202]: No worries, thanks for your review!

**Pull request merged**